tests/BasicEndToEndTest.cc

```diff
@@ -4277,7 +4277,9 @@ TEST(BasicEndToEndTest, testBatchReceiveClose) { testBatchReceiveClose(false); }
 
 TEST(BasicEndToEndTest, testBatchReceiveCloseWithMultiConsumer) { testBatchReceiveClose(true); }
 
-TEST(BasicEndToEndTest, testAckMsgList) {
+class AcknowledgeTest : public testing::TestWithParam<int> {};
```

Comment:
How about moving this test class to a separate file.

Reply:
It make sense to me.

BTW, the `BasicEndToEndTest` is too huge, adding a test into this file requires much more compile time than others. The original purpose is to use another PR to split tests in this file. For this PR, I will split the test class to another file soon.